src/app/components/utility/input/input.component.ts

```diff
@@ -235,7 +241,7 @@ export class InputComponent implements OnInit {
   }
 
   toggleErrorMessage () {
```

Comment:
Or, at least simplify the function to this:
`    return !(((this.isRequired && this.isEmpty) || (!this.isValid && !this.isEmpty) || this.message !== '') && this.isDirty);
`

Reply:
done